light/provider/provider.go

```diff
@@ -25,4 +25,7 @@ type Provider interface {
 
 	// ReportEvidence reports an evidence of misbehavior.
 	ReportEvidence(context.Context, types.Evidence) error
+
+	// Returns the IP address of the provider in string format
+	String() string
```

Comment:
Since this string has a specific meaning, I think we should use a more specific name, perhaps: `IP` or `Address` or `IPAddress`. A `String` method is generally used to describe the whole object, not just part of it.

Reply:
The problem with introducing a new function is that we have another implementation of the provider interface in internal/statesync (BlockProvider). I chose string as BlockProvider has already a String implementation and this way I did not have to touch code in another package. I can add an ID function to BlockProvider  and rename this (changes http_test as well) ?

Reply:
I think it's worth a little bit larger change, to get a clearer name. The main reason I hesitate about overloading `String` here is that method is used by other packages (especially `fmt`) to get a general-purpose representation of its receiver.